src/overlays/actors/ovl_Item_B_Heart/z_item_b_heart.c

```diff
@@ -10,26 +16,103 @@ void ItemBHeart_Destroy(Actor* thisx, PlayState* play);
 void ItemBHeart_Update(Actor* thisx, PlayState* play);
 void ItemBHeart_Draw(Actor* thisx, PlayState* play);
 
-/*
-const ActorInit Item_B_Heart_InitVars = {
-    ACTOR_ITEM_B_HEART,
-    ACTORCAT_BOSS,
-    FLAGS,
-    OBJECT_GI_HEARTS,
-    sizeof(ItemBHeart),
-    (ActorFunc)ItemBHeart_Init,
-    (ActorFunc)ItemBHeart_Destroy,
-    (ActorFunc)ItemBHeart_Update,
-    (ActorFunc)ItemBHeart_Draw
+void ItemBHeart_UpdateRotationAndScale(ItemBHeart* this, PlayState* play);
+
+const ActorInit Item_B_Heart_InitVars = { ACTOR_ITEM_B_HEART,
+                                          ACTORCAT_BOSS,
+                                          FLAGS,
+                                          OBJECT_GI_HEARTS,
+                                          sizeof(ItemBHeart),
+                                          (ActorFunc)ItemBHeart_Init,
+                                          (ActorFunc)ItemBHeart_Destroy,
+                                          (ActorFunc)ItemBHeart_Update,
+                                          (ActorFunc)ItemBHeart_Draw };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 0, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 800, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_STOP),
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_B_Heart/ItemBHeart_Init.s")
+void ItemBHeart_Init(Actor* thisx, PlayState* play) {
+    ItemBHeart* this = THIS;
+
+    if (Flags_GetCollectible(play, 0x1F)) {
+        Actor_Kill(&this->actor);
+    } else {
+        Actor_ProcessInitChain(&this->actor, sInitChain);
+        ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.8f);
```

Comment:
We prefer to use early returns for Actor_Kill

```suggestion
    if (Flags_GetCollectible(play, 0x1F)) {
        Actor_Kill(&this->actor);
        return;
    }
    Actor_ProcessInitChain(&this->actor, sInitChain);
    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.8f);
```